src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/taglets/TagletWriter.java

```diff
@@ -472,4 +473,15 @@ private boolean isLongOrHasComma(Content c) {
                 .replaceAll("\\R", "\n");             // normalize newlines
         return s.length() > TAG_LIST_ITEM_MAX_INLINE_LENGTH || s.contains(",");
     }
+
+    // Test if element is the same as or belongs to the current page element
+    private boolean isDifferentTypeElement(Element element) {
+        if (element.getKind().isClass() || element.getKind().isInterface()) {
```

Comment:
Your version might be a bit more readable, but this one is shorter:
```suggestion
        if (element.getKind().isDeclaredType()) {
```